### PR TITLE
bump up version and gate magic-wand version

### PR DESCRIPTION
**Pull request by dhuangnm**
1. bump up main to 0.4.0
2. gated nm-magic-wand version to ~=<nm-vllm version> for the release; nightly still installs the latest nm-magic-wand-nightly

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -458,6 +452,11 @@ def _read_requirements(filename: str) -> List[str]:
     python_requires=">=3.8",
     install_requires=get_requirements(),
     ext_modules=ext_modules,
+    # UPSTREAM SYNC: needed for sparsity
```
Comment (andy-neuma) [inline setup.py:455]: where is "version" being set?

File: setup.py [hunk 2]
```diff
@@ -411,12 +411,6 @@ def _read_requirements(filename: str) -> List[str]:
 if not _is_neuron():
     ext_modules.append(CMakeExtension(name="vllm._C"))
 
-# UPSTREAM SYNC: needed for sparsity
```
Comment (andy-neuma) [inline setup.py:414]: what is the reason to move this?

Comment (dhuangnm) [reply setup.py]: moved back :)

File: setup.py (hunk 1 above)
Comment (dhuangnm) [reply setup.py]: The version is getting from get_vllm_version() function which reads from __init__.py

File: setup.py [hunk 3]
```diff
@@ -458,6 +452,11 @@ def _read_requirements(filename: str) -> List[str]:
     python_requires=">=3.8",
     install_requires=get_requirements(),
     ext_modules=ext_modules,
+    # UPSTREAM SYNC: needed for sparsity
+    _sparsity_deps = [f"nm-magic-wand-nightly~={version}"]
+    nm_release_type = os.getenv(NM_RELEASE_TYPE)
+    if nm_release_type == 'RELEASE':
+        _sparsity_deps = [f"nm-magic-wand~={version}"]
```
[the remark this answers is not in the document]
Comment (dhuangnm) [reply setup.py]: yep good catch, fixed it

**Pull request merged**